azure/scope/machinepoolmachine.go

```diff
@@ -246,7 +247,13 @@ func (s *MachinePoolMachineScope) SetFailureReason(v capierrors.MachineStatusErr
 
 // ProviderID returns the AzureMachinePool ID by parsing Spec.FakeProviderID.
 func (s *MachinePoolMachineScope) ProviderID() string {
-	return s.AzureMachinePoolMachine.Spec.ProviderID
+	// Transform the VMSS instance resource representation to conform to the cloud-provider-azure representation
+	providerID, err := azureutil.ConvertResourceGroupNameToLower(s.AzureMachinePoolMachine.Spec.ProviderID)
+	if err != nil {
+		// return original providerID
+		return s.AzureMachinePoolMachine.Spec.ProviderID
```

Comment:
for all the other ProviderIDs everywhere, we make sure to convert the RG name to lowercase before we even set it on the Object Spec. Where is this getting set for AzureMachinePoolMachine.Spec? 

This approach works to fix the CAPZ controller, but modifying the ID that gets set directly on the object would be slightly better IMO because then the Node and the CAPZ CRD provider IDs would actually match when the user looks at them. 

Side note: I opened #2892 as a follow-up because I was reminded that @jackfrancis did the work to make it exportable and we never followed up to use it.

Reply:
While testing the change before opening this PR i did try to lowercase the provider ID for `AzureMachinePoolMachine` at write time. 

The reason why i decided to go for this is that if you had a cluster with the ongoing issue and updated the controller it would break even more since the controller would think that the 2 machines are different machines but it would fail to reconcile because the objects have the same name. 

With the change at read time updating the controller on a cluster experiencing the issue would cleanly fix and reconcile.

Maybe is not a real issue though since no one should be running a live production cluster with machinepools and a capitalized resourceGroup name ? 

I can definetely change it back to write time if you think that's the right way to do it

Reply:
@CecileRobertMichon i created a separate PR with the `fix on write` rather than the one on read 

https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2894

to keep things clean.

Reply:
Thanks @primeroz. IMO #2894 is a better fix long term so we should go with that, especially given 

> no one should be running a live production cluster with machinepools and a capitalized resourceGroup name

If we wanted to we could technically do both (fix the root cause at write time then also merge this to fix existing MPs but IMO that's a bit overkill),

@mboersma @devigned any thoughts?

Reply:
> If we wanted to we could technically do both (fix the root cause at write time then also merge this to fix existing MPs but IMO that's a bit overkill),

I agree with you , and it might just become more confusing in code going forward for someone else looking at the code with no context and wondering why it happens in 2 different spots. 

but up to you :)

Reply:
I think it's sufficient to fix it in one place, and #2894 is preferable.